src/EnergyPlus/UnitarySystem.cc

```diff
@@ -1886,6 +1886,44 @@ namespace UnitarySystems {
 
+            MSHPIndex = this->m_DesignSpecMSHPIndex;
+            if (MSHPIndex > -1) {
+                // set no load air flow ratio local var
+                Real64 NoLoadCoolingAirFlowRateRatio = 1.0;
+                Real64 NoLoadHeatingAirFlowRateRatio = 1.0;
+                if (designSpecMSHP[MSHPIndex].coolingVolFlowRatio[0] == DataSizing::AutoSize) {
+                    NoLoadCoolingAirFlowRateRatio = 1.0 / double(designSpecMSHP[MSHPIndex].numOfSpeedCooling);
+                } else {
+                    NoLoadCoolingAirFlowRateRatio = min(this->m_NoLoadAirFlowRateRatio, designSpecMSHP[MSHPIndex].coolingVolFlowRatio[0]);
+                }
+                if (designSpecMSHP[MSHPIndex].heatingVolFlowRatio[0] == DataSizing::AutoSize) {
+                    NoLoadHeatingAirFlowRateRatio = 1.0 / double(designSpecMSHP[MSHPIndex].numOfSpeedHeating);
+                } else {
+                    NoLoadHeatingAirFlowRateRatio = min(this->m_NoLoadAirFlowRateRatio, designSpecMSHP[MSHPIndex].heatingVolFlowRatio[0]);
+                }
+                this->m_NoLoadAirFlowRateRatio = min(NoLoadCoolingAirFlowRateRatio, NoLoadHeatingAirFlowRateRatio);
+            }
+            // variable speed dx coils need local variable for saving noload flow fraction
+            Real64 NoLoadFractionForVarSpeedCoils = 1.0;
+            if (this->m_CoolingCoilType_Num == DataHVACGlobals::Coil_CoolingAirToAirVariableSpeed || 
+                this->m_HeatingCoilType_Num == DataHVACGlobals::Coil_HeatingAirToAirVariableSpeed) {
+                Real64 NoLoadCoolingAirFlowRateRatio = 1.0;
+                Real64 NoLoadHeatingAirFlowRateRatio = 1.0;
+                if (this->m_CoolCoilExists && this->m_CoolingCoilType_Num == DataHVACGlobals::Coil_CoolingAirToAirVariableSpeed) {
+                    Real64 MaxSpeedFlowRate = VariableSpeedCoils::VarSpeedCoil(this->m_CoolingCoilIndex).MSRatedAirVolFlowRate(VariableSpeedCoils::VarSpeedCoil(this->m_CoolingCoilIndex).NumOfSpeeds);
+                    if (MaxSpeedFlowRate > 0.0) {
+                        NoLoadCoolingAirFlowRateRatio = VariableSpeedCoils::VarSpeedCoil(this->m_CoolingCoilIndex).MSRatedAirVolFlowRate(1) / MaxSpeedFlowRate;
+                    }
+                }
+                if (this->m_HeatCoilExists && this->m_HeatingCoilType_Num == DataHVACGlobals::Coil_HeatingAirToAirVariableSpeed) {
+                    Real64 MaxSpeedFlowRate = VariableSpeedCoils::VarSpeedCoil(this->m_HeatingCoilIndex).MSRatedAirVolFlowRate(VariableSpeedCoils::VarSpeedCoil(this->m_HeatingCoilIndex).NumOfSpeeds);
+                    if (MaxSpeedFlowRate > 0.0) {
+                        NoLoadHeatingAirFlowRateRatio = VariableSpeedCoils::VarSpeedCoil(this->m_HeatingCoilIndex).MSRatedAirVolFlowRate(1) / MaxSpeedFlowRate;
+                    }
+                }
+                NoLoadFractionForVarSpeedCoils = min(NoLoadCoolingAirFlowRateRatio, NoLoadHeatingAirFlowRateRatio);
```

Comment:
and store this fraction in this->m_NoLoadAirFlowRateRatio instead of NoLoadFractionForVarSpeedCoils

Reply:
Done. I was thinking that some of the other sizing method for no-load flow might need this variable but may be not.